dev/Telemetry/RuntimeProfiler.cpp

```diff
@@ -111,62 +174,58 @@ namespace RuntimeProfiler {
                     //  up on the next FireEvent().
                     continue;
                 }
-            
-                //  Zeroing out AND getting value.
-                cHits = InterlockedExchange(m_Counts[ii].pInstanceCount, 0);
-            
-                if (0 != cHits)
+
+                //  Zeroing out AND getting current value.
+                cHits = (UINT)(::InterlockedExchange(m_Counts[ii].pInstanceCount, 0));
+
+                if (0 == cHits)
                 {
-                    HRESULT     hr;
-                    
-                    //  We're using id's instead.  The entry in the list will
-                    //  look like '[type index|method index]:count'
-                    hr = StringCchPrintfExW(
-                            pszDest,
-                            cchDest,
-                            &pszDest,
-                            &cchDest,
-                            STRSAFE_NULL_ON_FAILURE,
-                            L"%ls[%d|%d]:%d",
-                            (bSeparator?L",":L""),
-                            (int)m_Counts[ii].uTypeIndex,
-                            (int)m_Counts[ii].uMethodIndex,
-                            cHits);
-
-                    if (S_OK == hr)
+                    //  Next!
+                    continue;
+                }
+
+                for (;;)
+                {
+                    PWSTR   pszWrite = pszLastEntry;
+
+                    //  If not first entry, enter a separator...
+                    if (pszWrite != pszOutput)
                     {
-                        cMethodsLogged++;
-                        bSeparator = true;
+                        pszWrite = WriteString(pszWrite, L",", pszLastChar);
                     }
-                    else if ((STRSAFE_E_INSUFFICIENT_BUFFER == hr) ||
-                             (STRSAFE_E_INVALID_PARAMETER == hr))
+
+                    //  The entry in the list will look like
+                    //    '[type index|method index]:count'
```

Comment:
Any reason this isn't using a standard format string like StringCchPrintf or any of its friends?